src/types.ts

```diff
@@ -304,14 +313,25 @@ export interface IEmitterOptions {
    * Set to true to enable local aws
    */
   isLocal?: boolean;
+  /**
+   * @description Provide this to override the default logger and control
+   * log levels and mapping
+   */
+  logger?: Logger;
 }
 
 export type DefaultQueueOptions = Omit<
   Topic,
   "separateConsumerGroup" | "isFifo" | "exchangeType"
 >;
 
-export type EventListener<T> = (...args: T[]) => Promise<void>;
+export interface MessageMetaData {
+  executionTraceId: string;
+  messageId?: string;
```

Comment:
messageId can be undefined? 🤔

Reply:
The message object of type IMessage declares id as optional and it is picked from there to pass into the listener function